pulsar-broker/src/test/java/org/apache/pulsar/broker/intercept/MangedLedgerInterceptorImplTest.java

```diff
@@ -256,6 +257,29 @@ public void testFindPositionByIndex() throws Exception {
         ledger.close();
     }
 
+    @Test
+    public void testBeforeAddEntryWithException() throws Exception {
```

Comment:
I wander how this test can verify the duplicate release

Reply:
There is no such error log with the bug fix in the test. Since the following error can not interrupt the normal execution, so it is hard to cover the error in a test case. Do you have any idea?
```
ERROR - [test-OrderedScheduler-1-0:SafeRunnable@38] - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, decrement: 1
```

Reply:
> There is no such error log with the bug fix in the test.

This will not prevent regression. The test will alwalys pass no matter duplicate release or not.

How about verifying the `Bytebuf.refCnt()` like below ?  @HQebupt 
```suggestion
  ByteBuf buffer = Unpooled.wrappedBuffer("message".getBytes());
        ManagedLedger ledger = factory.open(ledgerAndCursorName, config);

        CountDownLatch countDownLatch = new CountDownLatch(1);
        ledger.asyncAddEntry(buffer, new AsyncCallbacks.AddEntryCallback() {
            @Override
            public void addComplete(Position position, ByteBuf entryData, Object ctx) {
                countDownLatch.countDown();
            }

            @Override
            public void addFailed(ManagedLedgerException exception, Object ctx) {
                countDownLatch.countDown();
            }
        }, null);
        countDownLatch.await();
        assertEquals(buffer.refCnt(), 1);
```

Reply:
Great! I made a minor modification. PTAL @AnonHxy thanks.